doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,311 @@
+
+   Modify the configuration settings as follows, including a new firmware version:
+
+   .. code-block:: console
+
+      CONFIG_BUILD_S1_VARIANT=y
+      CONFIG_SB_SIGNING_KEY_FILE="/path/to/priv_c.pem"
+      CONFIG_FW_INFO_FIRMWARE_VERSION=2
+
+#. Deploy the firmware update.
+   Observe the bootloader checking the hashes of the public keys against the new image, then invalidating the earlier keys:
+
+   .. code-block:: console
+
+      *** Booting Zephyr OS build ...  ***
+      Attempting to boot slot 1.
+      Attempting to boot from address 0x84800.
+      Verifying signature against key 0.
+      Hash: 0xda...4f
+      Public key didn't match, try next.
+      Verifying signature against key 1.
+      Hash: 0x5c...f5
+      Public key didn't match, try next.
+      Verifying signature against key 2.
+      Hash: 0x19...73
+      Invalidating key 0.
+      Invalidating key 1.
+      Firmware signature verified.
+      Firmware version 2
+      Setting monotonic counter (version: 2, slot: 1)
+      *** Booting Zephyr OS build ...  ***
+      ...
+
+Testing
+-------
+
+You can test that the bootloader will no longer boot images signed with the earlier keys by uploading an image signed with one of them.
+To do this, recompile the application with the following options:
+
+.. code-block:: console
+
+   CONFIG_SB_SIGNING_KEY_FILE="/path/to/priv_b.pem"
+   CONFIG_FW_INFO_FIRMWARE_VERSION=3
+
+
+To facilitate testing, you can use ``nrfjprog`` to program this image directly into a slot:
+
+.. code-block:: console
+
+   nrfjprog -f nRF52 -r --verify --program build/zephyr/signed_by_b0_s0_image.hex --sectorerase
+
+Observe the bootloader catching the invalid image and booting the valid image in the other slot:
+
+.. code-block:: console
+
+   *** Booting Zephyr OS build ...  ***
+   Attempting to boot slot 0.
+   Attempting to boot from address 0x9000.
+   Firmware has been invalidated: 0x91020000.
+   Failed to validate, permanently invalidating!
+   Attempting to boot slot 1.
+   Attempting to boot from address 0x84800.
+   Key 0 has been invalidated, try next.
+   Key 1 has been invalidated, try next.
+   Verifying signature against key 2.
+   Hash: 0x19...73
+   Invalidating key 0.
+   Invalidating key 1.
+   Firmware signature verified.
+   Firmware version 2
+   *** Booting Zephyr OS build ...  ***
+   ...
+
+Recompiling with ``priv_c.pem`` and the incremented firmware version will correctly boot into the new image.
+
+.. _ug_fw_update_image_versions:
+
+Image versions
+**************
+
+With the |NCS|, versioning an image for use in firmware releases and updates depends on the type of bootloader that will be booting into the new firmware release.
+
+.. _ug_fw_update_image_versions_b0:
+
+|NSIB|
+======
+
+See the bootloader's :ref:`bootloader_monotonic_counter` documentation for more information.
```

Comment:
This link leads to a section that asks me to check CONFIG_SB_MONOTONIC_COUNTER docs and mentions two other Kconfig options. Wouldn't it be easier to provide the list of the Kconfig options here or actually move the bulk of the info from the sample docs to here?

Reply:
fixed